### PR TITLE
chore: CE changes for linting

### PR DESCRIPTION
**Pull request by Rishabh-Rathod**
#### PR fixes following issue(s)

Fixes # (issue number)

#### Type of change

- Chore (housekeeping or task changes that don't impact user perception)

- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/src/ce/plugins/Linting/lib/entity/types.ts [hunk 1]
```diff
@@ -1,16 +1,19 @@
...
-}
+export const ENTITY_TYPE = {
+  ACTION: "ACTION",
+  WIDGET: "WIDGET",
+  APPSMITH: "APPSMITH",
+  JSACTION: "JSACTION",
+};
+
+type ValueOf<T> = T[keyof T];
+export type EntityTypeValue = ValueOf<typeof ENTITY_TYPE>;
```
Comment (ApekshaBhosale) [inline app/client/src/ce/plugins/Linting/lib/entity/types.ts:11]: do we need this? isn't ENTITY_TYPE is value only? @Rishabh-Rathod

**Review by ApekshaBhosale — CHANGES_REQUESTED** [review]

File: app/client/src/ce/plugins/Linting/lib/entity/types.ts (hunk 1 above)
Comment (Rishabh-Rathod) [reply app/client/src/ce/plugins/Linting/lib/entity/types.ts:11]: Yes, ENTITY_TYPE is an object and cannot be used for type. Hence we are deriving type from the values of the object. 
`EntityTypeValue` will have "JSAction" | "ACTION" | "WIDGET" as its type.

Comment (ApekshaBhosale) [reply app/client/src/ce/plugins/Linting/lib/entity/types.ts:11]: @ankitakinger check this if `EntityTypeValue` is required to be written like that please once?

**Review by ApekshaBhosale — APPROVED** [review]

**Pull request merged**